### PR TITLE
docs: Add a contributing.md

### PR DESCRIPTION
**Pull request by knownasilya**
@acburdine any other suggestions?

I plan to add more later, about debugging and submitting PRs and issues.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+npm run build
+npm link dist
+```
+
+This makes denali available as `denali` and now you can use `denali new my-app`.
+After generating a new app, you need to link back to the built version of denali.
+
+```sh
+cd my-app
+npm link ../denali/dist
```
Comment (acburdine) [inline CONTRIBUTING.md]: assuming `npm link dist` has been run correctly, this should probably be `npm link denali` -> this particular command assumes that the app is in the same folder as denali itself, which could be confusing.

Comment (knownasilya) [inline CONTRIBUTING.md]: Good point.

**Comment by knownasilya** [comment]
I don't think it needs to be installed globally the first time, since denali-cli is present, and we use `npm run build`, which uses the internal `node_modules/.bin` directory to find "globals".

**Comment by rv-aburdine** [comment]
It's not about having `denali-cli` installed, as yeah that is included in denali's dependencies itself, but I'm talking about the actual `denali build` command, the code for which is here: https://github.com/denali-js/denali/blob/master/commands/build.ts . Because the command itself won't exist, then denali won't be able to be built.

**Comment by knownasilya** [comment]
@acburdine I added a couple more sections

**Comment by davewasmer** [comment]
Went ahead and merged this just to get something in there -we can tweak if needed.

**Pull request merged**